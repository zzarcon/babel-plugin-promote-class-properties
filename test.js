const pluginTester = require('babel-plugin-tester');
const promoteClassProperties = require('.');

const tests = [{
  snapshot: true,
  title: 'basic',
  code: `
    class Foo {
      constructor() {
        this.bar = this.bar.bind(this);
      }

      bar() {
        const miau = 1;
      }
    }
  `
}, {
  snapshot: true,
  title: 'method with args',
  code: `
    class Foo {
      constructor() {
        this.methodWithArgs = this.methodWithArgs.bind(this);
      }

      methodWithArgs(arg1, arg2) {
        console.log(arg1, arg2)
      }
    }
  `
}, {
  snapshot: false,
  title: 'not binded',
  code: `
    class Foo {
      constructor() {
        this.notBinded = this.notBinded();
      }

      notBinded() {
        return 1;
      }
    }
  `
}, {
  snapshot: false,
  title: 'bind outer method',
  code: `
    const outerMethod = () => {};

    class Foo {
      constructor() {
        this.outerMethod = outerMethod.bind(this);
      }

      outerMethod() {
        return 1;
      }
    }
  `
}];

pluginTester({
  plugin: promoteClassProperties,
  tests
});
