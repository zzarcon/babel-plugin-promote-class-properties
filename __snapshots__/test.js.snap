// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"
class Foo {
  constructor() {
    this.bar = this.bar.bind(this);
  }

  bar() {
    const miau = 1;
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

class Foo {
  constructor() {}

  bar = () => {
    const miau = 1;
  };
}
"
`;

exports[`method before constructor 1`] = `
"
class Foo {
  bar() {
    console.log(1)
  }

  constructor() {
    this.bar = this.bar.bind(this);
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

class Foo {
  bar() {
    console.log(1);
  }

  constructor() {}
}
"
`;

exports[`method with args 1`] = `
"
class Foo {
  constructor() {
    this.methodWithArgs = this.methodWithArgs.bind(this);
  }

  methodWithArgs(arg1, arg2) {
    console.log(arg1, arg2)
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

class Foo {
  constructor() {}

  methodWithArgs = (arg1, arg2) => {
    console.log(arg1, arg2);
  };
}
"
`;
